main.go

```diff
@@ -202,22 +225,34 @@ func buildParseErrorString(target, e string, err error) string {
 	return msg
 }
 
+func SetupDeferredAccessLogging(accessLogger *zap.Logger, accessLogDetails *carbonapipb.AccessLogDetails, t time.Time, logAsError bool)  {
```

Comment:
Also name of the function is wrong - you are not setting up logging, but actually do log.